main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -369,6 +369,19 @@ object Verifier {
         case AtomicOp.Cast =>
           tpe
 
+        case AtomicOp.Region =>
+          check(expected = MonoType.Region)(actual = tpe, loc)
+
+        case AtomicOp.Spawn =>
+          val List(t1, t2) = ts
+          t1 match {
+            case MonoType.Arrow(List(MonoType.Unit), _) => ()
```

Comment:
In the other PR you also check for a `Unit -> _` functions, you should choose one method and do it the same way both places.